framework/include/outputs/JsonIO.h

```diff
@@ -12,7 +12,16 @@
 #include "nlohmann/json.h"
 
 class MooseApp;
+namespace libMesh
+{
+class Point;
+}
 
 // Specializations for to_json, which _must_ be specialized in the namespace
 // in which the object is found
 void to_json(nlohmann::json & json, const MooseApp & app); // MooseDocs:to_json
+
+namespace libMesh
+{
+void to_json(nlohmann::json & json, const Point & p);
+}
```

Comment:
```suggestion
void to_json(nlohmann::json & json, const libMesh::Point & p);
```

Reply:
I think this is needed actually.
See comment above, needs to be instantiated in namespace it's used

Reply:
Alright. Can you add to the comment and say 
```
// Specializations for to_json, which _must_ be specialized in the namespace                                          
// in which the object is found in order to enable argument-dependent lookup
```

Reply:
And it would also be helpful to add a comment in `ReporterContext::store` about how `nlohmann` qualification is needed because `std::vector` specialization is *not* in the `std` namespace, it's in the `nlohmann` namespace. (I know this is not your work, but this was pretty confusing as i was trying to understand why my suggested change was not working). I could also do this in a separate PR if you would prefer.

Reply:
For your first comment, this already exists: https://github.com/idaholab/moose/blob/2a048be44daac9b4c7fb3f3c265284df72987408/framework/include/outputs/JsonIO.h#LL16C32-L16C32

> // Specializations for to_json, which _must_ be specialized in the namespace
> // in which the object is found

Reply:
Yea it needs more meaning. Just saying it has to be done is not enough for me. It's nice to know why

Reply:
You'll notice that I built my addition off of your comment

Reply:
So in full I would have the comment be
```
// Overloads for to_json, which _must_ be overloaded in the namespace                                          
// in which the object is found in order to enable argument-dependent lookup
// See https://en.cppreference.com/w/cpp/language/adl for more information
```
I've also clarified the language to "overload" instead of "specialization" since these are not specializations in the C++ terminology

Reply:
Sounds good, just wanted to make sure there wasn't a duplicate